codec/any_test.go

```diff
@@ -28,38 +27,29 @@ func TestMarshalAny(t *testing.T) {
 	cdc := codec.NewProtoCodec(registry)
 
 	kitty := &testdata.Cat{Moniker: "Kitty"}
-	bz, err := codec.MarshalAny(cdc, kitty)
+	bz, err := cdc.MarshalInterface(kitty)
 	require.NoError(t, err)
 
 	var animal testdata.Animal
 
 	// empty registry should fail
-	err = codec.UnmarshalAny(cdc, &animal, bz)
+	err = cdc.UnmarshalInterface(bz, &animal)
 	require.Error(t, err)
 
 	// wrong type registration should fail
 	registry.RegisterImplementations((*testdata.Animal)(nil), &testdata.Dog{})
-	err = codec.UnmarshalAny(cdc, &animal, bz)
+	err = cdc.UnmarshalInterface(bz, &animal)
 	require.Error(t, err)
 
 	// should pass
 	registry = NewTestInterfaceRegistry()
 	cdc = codec.NewProtoCodec(registry)
-	err = codec.UnmarshalAny(cdc, &animal, bz)
+	err = cdc.UnmarshalInterface(bz, &animal)
 	require.NoError(t, err)
 	require.Equal(t, kitty, animal)
 
 	// nil should fail
 	registry = NewTestInterfaceRegistry()
-	err = codec.UnmarshalAny(cdc, nil, bz)
+	err = cdc.UnmarshalInterface(bz, nil)
 	require.Error(t, err)
 }
-
-func TestMarshalAnyNonProtoErrors(t *testing.T) {
```

Comment:
why do we remove this test?

Reply:
Because this test doesn't make sense any more. `cdc.MarshalInterface` takes `proto.Message` as a parameter. Previously it was taking `interface{}` and failing in the runtime if a type (like here: `int`, in the test) doesn't implement that interface. It's better to have that check during compilation time.